.github/workflows/mainPipeline.yml

```diff
@@ -0,0 +1,147 @@
+        
+      - name: Cache SonarCloud packages
+        uses: actions/cache@v1
+        with:
+          path: ~\sonar\cache
+          key: ${{ runner.os }}-sonar
+          restore-keys: ${{ runner.os }}-sonar
+          
+      - name: Cache SonarCloud scanner
+        id: cache-sonar-scanner
+        uses: actions/cache@v1
+        with:
+          path: .\.sonar\scanner
+          key: ${{ runner.os }}-sonar-scanner
+          restore-keys: ${{ runner.os }}-sonar-scanner
+          
+      - name: Install SonarCloud scanner
+        if: steps.cache-sonar-scanner.outputs.cache-hit != 'true'
+        shell: powershell
+        run: |
+          New-Item -Path .\.sonar\scanner -ItemType Directory
+          dotnet tool update dotnet-sonarscanner --tool-path .\.sonar\scanner
+          
+      - name: Run tests
+        shell: powershell
+        run: dotnet test Selenium.Axe/Selenium.Axe.Test/Selenium.Axe.Test.csproj --results-directory "${{ github.workspace }}/artifactTests/coverage/testResults/" /p:CoverletOutput="${{ github.workspace }}/artifactTests/coverage/testResults/" /p:CollectCoverage=true /p:CoverletOutputFormat=opencover -c Release -v D
+      
+      - name: Build and analyze
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}  # Needed to get PR information, if any
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+        shell: powershell
+        run: | 
+          .\.sonar\scanner\dotnet-sonarscanner begin /k:"TroyWalshProf_SeleniumAxeDotnet" /o:"troywalshprof"  /d:sonar.login="${{ secrets.SONAR_TOKEN }}" /d:projectVersion="${{ steps.ver.outputs.BUILD_NUMBER }}"  /d:sonar.host.url="https://sonarcloud.io" /d:sonar.cs.opencover.reportsPaths="${{ github.workspace }}/artifactTests/coverage/testResults/*.opencover.xml"  /d:sonar.cs.dotcover.reportsPaths="${{ github.workspace }}/artifactTests/coverage/testResults/*.coverage"
+          dotnet build Selenium.Axe/Selenium.Axe.sln  --configuration Release -p:Version=${{ steps.ver.outputs.BUILD_NUMBER }} -o "${{github.workspace}}/artifactTests/packages"
+          .\.sonar\scanner\dotnet-sonarscanner end /d:sonar.login="${{ secrets.SONAR_TOKEN }}"
+ 
+      - uses: actions/upload-artifact@v2
+        if: always()
+        name: "Upload NuGet packages artifact"
+        with:
+          name: packages
+          path: |
+            ${{github.workspace}}/artifactTests/packages/**/*.nupkg
+            ${{github.workspace}}/artifactTests/packages/**/*.snupkg
+   
+      - uses: actions/upload-artifact@v2
+        if: always()
+        name: "Upload test artifact"
+        with:
+          name: testResults
+          path: |
+            ${{ github.workspace }}/artifactTests/coverage/testResults/**/*.xml
+            ${{ github.workspace }}/artifactTests/coverage/testResults/
+
+  publish:
+    name: Publish NuGet - v${{ needs.buildTest.outputs.buildNumber }}
+    if: github.event_name == 'push' || github.event_name == 'workflow_dispatch'
```

Comment:
I worry that doing this on every push to main makes it easy to accidentally issue patch releases (the release-drafter default) for things that aren't really patch-level changes. Maybe this should be paired with some sort of check that forbids merging PRs that haven't been explicitly labelled as major/minor/patch, and/or some mechanism for non-user-facing changes that shouldn't trigger releases?

Reply:
This is why we have a protection rule on the nuget environment.
This way every PR is treated as a release candidate and it is up to the reviews if they want to allow the push to nuget.org or not.

Somewhat related, how would you feel about being one of the approvers?

Reply:
Here is how it will look once me get this merged into main:
![image](https://user-images.githubusercontent.com/2066107/119181044-b6a1f600-ba36-11eb-8328-d6df2ac1e3f1.png)
*From a different project that does the same thing

Reply:
Ah, got it, I didn't realize that setting the environment property would have that effect. Neat!

Reply:
> Somewhat related, how would you feel about being one of the approvers?

This is fine by me!